### PR TITLE
Remove residual Python 2 specific code

### PR DESCRIPTION
**Pull request by EwoutH**
This commit cleans up the codebase by removing lingering Python 2 specific constructs.

**Update:** After #5824 did most of the work, the changes remaining in this PR are very small.

### CODE REVIEW

**Comment by scoder** [comment]
Hmm. Thanks. We already have a PR coming up for this:
https://github.com/cython/cython/pull/5824

Need to check if there are any relevant differences.

**Comment by EwoutH** [comment]
Didn't see that one! I'm fine with doing #5824 first, and then seeing what remains of this PR.

**Comment by EwoutH** [comment]
I'm going to stack some more changes onto this.

**Comment by EwoutH** [comment]
I made quite some changes to the ExprNodes module, so I split them of into a separate PR: https://github.com/cython/cython/pull/5834

File: Cython/Compiler/ExprNodes.py [hunk 1]
```diff
@@ -13,8 +13,7 @@
                Builtin=object, Symtab=object, Utils=object, find_coercion_error=object,
                debug_disposal_code=object, debug_temp_alloc=object, debug_coercion=object,
                bytearray_type=object, slice_type=object, memoryview_type=object,
-               builtin_sequence_types=object, _py_int_types=object,
-               IS_PYTHON3=cython.bint)
+               builtin_sequence_types=object, _py_int_types=object)
```
Comment (scoder) [inline Cython/Compiler/ExprNodes.py:16]: The two different int types are gone as well.
```suggestion
               builtin_sequence_types=object)
```

File: Cython/Compiler/Builtin.py [hunk 2]
```diff
@@ -193,10 +193,6 @@ def declare_in_type(self, self_type):
     #('map',       "",     "",      ""),
     #('max',       "",     "",      ""),
     #('min',       "",     "",      ""),
-    BuiltinFunction('next',       "O",    "O",     "__Pyx_PyIter_Next",
-                    utility_code = iter_next_utility_code),   # not available in Py2 => implemented here
-    BuiltinFunction('next',      "OO",    "O",     "__Pyx_PyIter_Next2",
-                    utility_code = iter_next_utility_code),  # not available in Py2 => implemented here
```
Comment (scoder) [inline Cython/Compiler/Builtin.py:199]: These are inline functions for performance reasons. Please keep them.

File: Cython/Compiler/TreePath.py [hunk 3]
```diff
@@ -11,10 +11,7 @@
 import operator
 import sys
 
-if sys.version_info[0] >= 3:
-    _unicode = str
-else:
-    _unicode = unicode
+_unicode = str
```
Comment (scoder) [inline Cython/Compiler/TreePath.py:14]: This is worth removing (and "inlining") as well.

Comment (EwoutH) [reply Cython/Compiler/TreePath.py:14]: Could you (suggest how to) implement that?

File: Cython/Compiler/Builtin.py (hunk 2 above)
Comment (EwoutH) [reply Cython/Compiler/Builtin.py]: Reverted these changes (in 4784987)

**Comment by EwoutH** [comment]
@scoder Thanks for reviewing and the suggested. I applied them, there is one item left.

File: Cython/Compiler/TreePath.py (hunk 3 above)
Comment (da-woods) [reply Cython/Compiler/TreePath.py:14]: I think just a find/replace in your favourite text editor for `_unicode` -> `str`

Comment (EwoutH) [reply Cython/Compiler/TreePath.py:14]: So it says `str = str`?

Comment (EwoutH) [reply Cython/Compiler/TreePath.py]: It was even only one occurrence (see 9fff260), here's the full diff:

```diff
diff --git a/Cython/Compiler/TreePath.py b/Cython/Compiler/TreePath.py
index b1371c249a5..007581890d9 100644
--- a/Cython/Compiler/TreePath.py
+++ b/Cython/Compiler/TreePath.py
@@ -11,8 +11,6 @@
 import operator
 import sys
 
-_unicode = str
-
 path_tokenizer = re.compile(
     r"("
     r"'[^']*'|\"[^\"]*\"|"
@@ -165,7 +163,7 @@ def select(result):
                     continue
                 if attr_value == value:
                     yield attr_value
-                elif (isinstance(attr_value, bytes) and isinstance(value, _unicode) and
+                elif (isinstance(attr_value, bytes) and isinstance(value, str) and
                         attr_value == value.encode()):
                     # allow a bytes-to-string comparison too
                     yield attr_value
```

**Comment by EwoutH** [comment]
All comments resolved, should be good for final review and to go.

**Comment by scoder** [comment]
Thanks

**Pull request merged**